### PR TITLE
JAVA-3013 Updates to other components to address additional 4.x CVEs

### PR DESCRIPTION
**Pull request by absurdfarce**
Because apparently I have problems naming my branches correctly 🤦

### CODE REVIEW

File: core-shaded/pom.xml [hunk 1]
```diff
@@ -331,7 +331,7 @@
                   <!--
                   5) Don't import packages imported by shaded classes, if they are not used by the
                   driver bundle.
-                  -->!com.google.protobuf.*, !com.jcraft.jzlib.*, !com.ning.compress.*, !lzma.sdk.*, !net.jpountz.xxhash.*, !org.bouncycastle.*, !org.conscrypt.*, !org.apache.commons.logging.*, !org.apache.log4j.*, !org.apache.logging.log4j.*, !org.eclipse.jetty.*, !org.jboss.marshalling.*, !sun.misc.*, !sun.security.*, !com.barchart.udt.*, !com.fasterxml.aalto.*, !com.sun.nio.sctp.*, !gnu.io.*, !org.xml.sax.*, !org.w3c.dom.*, *
+                  -->!com.google.protobuf.*, !com.jcraft.jzlib.*, !com.ning.compress.*, !lzma.sdk.*, !net.jpountz.xxhash.*, !org.bouncycastle.*, !org.conscrypt.*, !org.apache.commons.logging.*, !org.apache.log4j.*, !org.apache.logging.log4j.*, !org.eclipse.jetty.*, !org.jboss.marshalling.*, !sun.misc.*, !sun.security.*, !com.barchart.udt.*, !com.fasterxml.aalto.*, !com.sun.nio.sctp.*, !gnu.io.*, !org.xml.sax.*, !org.w3c.dom.*, !com.aayushatharva.brotli4j.*, !com.github.luben.zstd.*, *
```
Comment (absurdfarce) [inline core-shaded/pom.xml:334]: [Netty 4.1.66](https://netty.io/news/2021/07/16/4-1-66-Final.html) introduced support for two new compression impls, Brotli + zstd.  The third-party impls of these compression algorithms had to be excluded here in order to get the OSGi tests to pass again.

This should've been included in the PR for JAVA-2977.  The OSGi integration tests don't run clean there without this change.

File: core/src/main/java/com/datastax/dse/driver/internal/core/graph/DseGraphTraversal.java [hunk 2]
```diff
@@ -33,16 +33,6 @@ public DseGraphTraversal(AsyncGraphResultSet firstPage) {
...
-  @Override
-  @SuppressWarnings("deprecation")
-  public org.apache.tinkerpop.gremlin.process.remote.traversal.RemoteTraversalSideEffects
-      getSideEffects() {
-    // This was deprecated as part of TINKERPOP-2265
-    // and is no longer being promoted as a feature.
-    // return null but do not throw "NotSupportedException"
-    return null;
-  }
-
```
Comment (absurdfarce) [inline core/src/main/java/com/datastax/dse/driver/internal/core/graph/DseGraphTraversal.java:45]: [Officially removed](https://tinkerpop.apache.org/docs/current/upgrade/#_deprecation_removal) in Tinkerpop 3.5.0

File: integration-tests/src/test/java/com/datastax/dse/driver/api/core/graph/remote/ClassicGraphTraversalRemoteIT.java [hunk 3]
```diff
@@ -70,14 +70,12 @@ protected boolean isGraphBinary() {
...
-    return AnonymousTraversalSource.traversal()
-        .withRemote(DseGraph.remoteConnectionBuilder(session()).build());
+    return traversal().withRemote(DseGraph.remoteConnectionBuilder(session()).build());
   }
 
   @Override
   protected SocialTraversalSource socialTraversalSource() {
-    return EmptyGraph.instance()
-        .traversal(SocialTraversalSource.class)
+    return traversal(SocialTraversalSource.class)
```
Comment (absurdfarce) [inline integration-tests/src/test/java/com/datastax/dse/driver/api/core/graph/remote/ClassicGraphTraversalRemoteIT.java:78]: AnonymousTraversalSouce is [preferred](https://tinkerpop.apache.org/docs/current/upgrade/#_anonymoustraversalsource) over EmptyGraph in 3.5.0.  Once the static import came in the additional refactoring above followed naturally.

File: manual/core/integration/README.md [hunk 4]
```diff
@@ -562,6 +562,7 @@ Here are the recommended TinkerPop versions for each driver version:
 
 <table>
 <tr><th>Driver version</th><th>TinkerPop version</th></tr>
+<tr><td>4.14.1</td><td>3.5.3</td></tr>
```
Comment (absurdfarce) [inline manual/core/integration/README.md:565]: It seems a little strange for Tinkerpop to get a minor version bump for only a patch release for the Java driver... not sure what I think about this yet.

File: core/revapi.json [hunk 5]
```diff
@@ -5539,6 +5539,1284 @@
         "old": "missing-class org.apache.tinkerpop.gremlin.structure.VertexProperty",
         "new": "missing-class org.apache.tinkerpop.gremlin.structure.VertexProperty",
         "justification": "JAVA-2907: switched Tinkerpop dependency to optional"
+      },
```
Comment (absurdfarce) [inline core/revapi.json:5542]: API changes brought on by the Jackson upgrade... apparently some of the class hierarchy was re-worked, especially around the exceptions.

**Comment by absurdfarce** [comment]
I'd still like to test this against the Quarkus plugin just to make sure the Graal upgrade didn't do any damage but I don't anticipate a major problem there.

File: manual/core/integration/README.md (hunk 4 above)
Comment (adutra) [reply manual/core/integration/README.md:565]: Hmm indeed. But if the goal is to fix CVEs I guess we don't have much of a choice. (Besides we are doing the same for Jackson).

**Review by adutra — APPROVED** [review]
LGTM. I was impressed by the number of revapi exceptions you had to make but I'm OK with it. Also, thank you for trying the OSGi tests, we tend to forget those :-)

**Comment by absurdfarce** [comment]
Just tried a working Graal native image app with both 4.14.0 and this code; both versions worked without issue.  This was against GraalVM 21.1, a lesser version than the graal-sdk version in his commit.  So we appear to be fine on that front.

**Pull request merged**